dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaIncomingConnector.java

```diff
@@ -96,6 +98,7 @@ private void shutdown() {
             String topic = config.getOptionalValue(KafkaConnectorConstants.TOPIC, String.class).orElse(channelName);
             int maxPollRecords = config.getOptionalValue(KafkaConnectorConstants.MAX_POLL_RECORDS, Integer.class).orElse(500);
             int unackedLimit = config.getOptionalValue(KafkaConnectorConstants.UNACKED_LIMIT, Integer.class).orElse(maxPollRecords);
+            int retrySeconds = config.getOptionalValue(KafkaConnectorConstants.CREATION_RETRY_SECONDS, Integer.class).orElse(0);
```

Comment:
In SmallRye, the wait between the retry is 30s. https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2/kafka/kafka.html I think the default should be 30s instead of 0.

Reply:
I disagree.

We can't tell the difference between a failure caused by the hostname not being resolvable and a failure caused by bad configuration.

If the user has configured the application wrongly, we want to app to fail to start as soon as possible, not pointlessly retry for 30s.

As far as we've seen so far, having the broker hostname not be resolvable is the only failure that Kafka treats this way, in most cases network failures are already retried automatically by the kafka client.

Reply:
No-one should have to use this unless they're deploying in an environment where the kafka broker and the app are being brought up at the same time. (e.g. during a kubernetes deployment in a test environment)